### DISCUSSION TITLE
Fields Searching

### DISCUSSION THREAD

**Discussion by UromPLAt** [nicotine-plus/nicotine-plus#2476] [Q&A] [answered]
Hello, is there a way to search songs by a specific field?

In example:
Artist:Burnham
Title:Can't Handle This
Filename:Handle AND Artist:Burnham

I couldn't find any information on this, searched in both GitHub discussions and issues
Thanks

**Comment by slook** [comment, marked as answer]
Unfortunately the network protocol does not transmit tag data such as 'Artist' nor 'Title' etc with the search result listings. It is only possible to search for the text in the file path that is the combined folder and file name.

In Nicotine+ the Result Filters are offered as a way to refine the results, but the text include/exclude fields operate on the entire path, rather that separating folder/file, so it is not always possible to drill down by artist.

Issue #1023 suggests folder (i.e. artist/album) level filtering, but it is unclear exactly how this would be implemented.

**Reply by UromPLAt** [reply]
ha, wondering if it was a business decision (copyright stuff) or a technical decision by Nir.
Interesting, thanks

**Reply by slook** [reply]
It is plausible that the unencrypted transmission of identifier tags between network peers might be intercepted by a malicious third party or bad actor, and as such this would be a good reason why the feature was never included in the protocol specification.

**Reply by mathiascode** [reply]
Since the Soulseek protocol was created in the early 2000s, I find it more likely that using the file path for identification was easier to develop and less resource intensive than scanning and storing all tags.
